python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -635,10 +635,14 @@ def _flush(self, objs):
 
         for obj in objs:
             with obj.lock:
-                obj._getWriteAccess()
-                obj_id = getattr(obj, _reg_id_str)
+                # check if the object is dirty, if not do nothing
                 if not obj._dirty:
                     continue
+
+                # flush the object. Need to call _getWriteAccess for consistency reasons
+                # TODO: getWriteAccess should only 'get write access', as that's not needed should it be called here?
+                obj._getWriteAccess()
+                obj_id = getattr(obj, _reg_id_str)
                 self.repository.flush([obj_id])
                 obj._dirty = False
                 self.repository.unlock([obj_id])
```

Comment:
This leaves an object loaded in memory in ganga and releases the file locks.
If someone/something/some-event causes the file on disk to change we lose all knowledge of it as we make no attempt to re-load the object.

Worst case scenario this allows for another instance of ganga to grab the job, manipulate it and then this instance to ignore the changes and obliterate them upon a future write action.

If we remove this lock we should fully un-load the job and we currently have no mechanism for this.

This just allows for 2 instances of ganga (launched on purpose or by accident by a user) to get into a very bad state which can be avoided by only releasing objects on exit at the moment.